Fix failures due to shortened wait time for docker start

For testing and the web frontend, the wait time for starting the container is set to 1s only. However, that sometimes doesn't seem to be enough.

Can we just do a `while` loop (or even just an if statement for possible longer sleep time if we run into this here)?

```
         logger.debug(f"Starting container with command: %s", shlex.join(startup_cmd))
        container = subprocess.Popen(
            startup_cmd,
            stdin=PIPE,
            stdout=PIPE,
            stderr=STDOUT,
            text=True,
            bufsize=1, # line buffered
        )
        time.sleep(START_UP_DELAY)
        # try to read output from container setup (usually an error), timeout if no output
        output = read_with_timeout(container, lambda: list(), timeout_duration=2)
        if output:
            logger.error(f"Unexpected container setup output: {output}")
        # Get the process IDs of the container
        # There should be at least a head process and possibly one child bash process
        bash_pids, other_pids = get_background_pids(container_obj)
        bash_pid = 1
        if len(bash_pids) == 1:
            bash_pid = bash_pids[0][0]
        elif len(bash_pids) > 1 or len(other_pids) > 0:
>           raise RuntimeError(f"Detected alien processes attached or running. Please ensure that no other agents are running on this container. PIDs: {bash_pids}, {other_pids}")
E           RuntimeError: Detected alien processes attached or running. Please ensure that no other agents are running on this container. PIDs: [['65', 'bash'], ['79', 'bash'], ['94', 'bash'], ['108', 'bash']], []
```

--- a/sweagent/environment/utils.py
+++ b/sweagent/environment/utils.py
@@ -322,6 +322,13 @@ def _get_persistent_container(ctr_name: str, image_name: str, persistent: bool =
     # Get the process IDs of the container
     # There should be at least a head process and possibly one child bash process
     bash_pids, other_pids = get_background_pids(container_obj)
+    total_time_slept = START_UP_DELAY
+    while len(bash_pids) > 1 or len(other_pids) > 0:
+        time.sleep(1)
+        total_time_slept += 1
+        bash_pids, other_pids = get_background_pids(container_obj)
+        if total_time_slept > 5*START_UP_DELAY:
+            break
     bash_pid = 1
     if len(bash_pids) == 1:
         bash_pid = bash_pids[0][0]

